83.md

```diff
@@ -0,0 +1,120 @@
+  "sig": "f644529abdf7ea12c570800847ccc337201fe6c47ea8e09902017e04d6f87605c4c3ab7cece1189df9271a27df06e0da0c6f35375098004644d4dfbc38ba78e7"
+}
+```
+
+```json
+{
+  "kind": 8395,
+  "id": "3047567edd9d74f694c648850fef128963973379be6a42f49d40c90524fbc079",
+  "pubkey": "c5dce01ee61fc62f62e2a825e2d598a839653b3175e0dcc072b1fe3c885f84a7",
+  "created_at": 1709847680,
+  "tags": [],
+  "content": "/**\n * Infinite async generator. Iterates messages pushed to it until closed.\n * Only one consumer is expected to use a Machina instance at a time.\n *\n * @example\n * ```ts\n * // Create the Machina instance\n * const machina = new Machina<string>();\n *\n * // Async generator loop\n * async function getMessages() {\n *   for await (const msg of machina.stream()) {\n *     console.log(msg);\n *   }\n * }\n *\n * // Start the generator\n * getMessages();\n *\n * // Push messages to it\n * machina.push('hello!');\n * machina.push('whats up?');\n * machina.push('greetings');\n * ```\n */\nexport class Machina<T> {\n  #queue: T[] = [];\n  #resolve: (() => void) | undefined;\n  #aborted = false;\n\n  constructor(signal?: AbortSignal) {\n    if (signal?.aborted) {\n      this.abort();\n    } else {\n      signal?.addEventListener('abort', () => this.abort(), { once: true });\n    }\n  }\n\n  /** Get messages as an AsyncGenerator. */\n  async *[Symbol.asyncIterator](): AsyncGenerator<T> {\n    while (!this.#aborted) {\n      if (this.#queue.length) {\n        yield this.#queue.shift() as T;\n        continue;\n      }\n\n      await new Promise<void>((_resolve) => {\n        this.#resolve = _resolve;\n      });\n    }\n\n    throw new DOMException('The signal has been aborted', 'AbortError');\n  }\n\n  /** Push a message into the Machina instance, making it available to the consumer of `stream()`. */\n  push(data: T): void {\n    this.#queue.push(data);\n    this.#resolve?.();\n  }\n\n  /** Stops streaming and throws an error to the consumer. */\n  private abort(): void {\n    this.#aborted = true;\n    this.#resolve?.();\n  }\n}\n",
+  "sig": "b01ab6e2273bc33594dde5634c68add95c6273b8a864bf8f4c8a85564db3e3df2c9981a3c5d8694cc4fa0e0f984635d6b51d5ca352eff96b7cb789c39429d303"
+}
+```
+
+## Immutability
+
+Source code events are considered immutable, and should NOT be deleted by supported clients or relays in response to kind `5` deletion requests.
```

Comment:
I agree about immutability, but not being able to delete it seems a bit overkill 😅